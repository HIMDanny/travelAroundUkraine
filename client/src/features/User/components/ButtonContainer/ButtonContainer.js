import React from 'react';
import { Box, Button } from '@mui/material';
import { useDispatch } from 'react-redux';
import { setIsModalOpen } from '../../../../store/slices/userSlice/userSlice';

const buttonBoxSX = { display: 'flex', width: '100%', justifyContent: 'end', my: '20px' };

const buttonSX = {
  width: { xs: '86px', mobile: '110px', tablet: '140px', laptop: '150px' },
  px: 0,
  py: { xs: '8px', mobile: '12px', desktop: '14px' },
  mx: 4,
  fontSize: { xs: '12px', laptop: '14px' },
};

const ButtonContainer = () => {
  const dispatch = useDispatch();

  return (
    <Box sx={buttonBoxSX}>
      <Button onClick={() => dispatch(setIsModalOpen(true))} sx={buttonSX}>
        log out
      </Button>
    </Box>
  );
};

export default ButtonContainer;
