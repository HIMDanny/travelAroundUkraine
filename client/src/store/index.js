import { configureStore } from '@reduxjs/toolkit';
import userReducer from './slices/userSlice/userSlice';
import tourReducer from './slices/tourSlice/tourSlice';
import catalogueReducer from './slices/catalogueSlice/catalogueSlice';
import cartReducer from './slices/cartSlice/cartSlice';
import orderSlice from './slices/orderSlice/orderSlice';
import inFavoritesReducer from './slices/inFavoritesSlice/inFavoritesSlice';

const store = configureStore({
  reducer: {
    userReducer,
    tour: tourReducer,
    catalogue: catalogueReducer,
    cart: cartReducer,
    order: orderSlice,
    favorites: inFavoritesReducer,
  },
});

export default store;
