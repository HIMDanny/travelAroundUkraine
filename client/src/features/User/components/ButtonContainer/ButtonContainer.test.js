import { render } from '@testing-library/react';
import ButtonContainer from './ButtonContainer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../../../store';

jest.mock('@mui/material/Button', () => ({ children }) => <button>{children}</button>);
jest.mock('@mui/material/Box', () => ({ children }) => <div>{children}</div>);

describe('Render ButtonContainer', () => {
  test('should ButtonContainer render', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <ButtonContainer />
        </BrowserRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
