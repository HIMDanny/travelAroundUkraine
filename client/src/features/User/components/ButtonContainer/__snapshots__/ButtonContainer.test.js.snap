// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ButtonContainer should ButtonContainer render 1`] = `
<DocumentFragment>
  <div>
    <button>
      log out
    </button>
  </div>
</DocumentFragment>
`;
